erpnext/public/js/controllers/taxes_and_totals.js

```diff
@@ -754,5 +754,61 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 				}
 			});
 		})
+	},
+	set_and_update_cultivation_tax: function () {
+		if (["Sales Order", "Sales Invoice", "Delivery Note", "Stock Entry"].includes(this.frm.doctype)) {
+			if (!this.frm.doc.license) {
+				return;
+			}
+		}
+
+		const me = this;
+		frappe.db.get_value("Compliance Info", { "name": this.frm.doc.license }, "license_for", (r) => {
+			if (["Sales Order", "Sales Invoice", "Delivery Note", "Stock Entry"].includes(this.frm.doctype)) {
+				console.log("test");
+				if (!r || r.license_for !== "Distributor") {
+					return;
+				}
+			}
+
+			frappe.call({
+				method: "erpnext.compliance.taxes.set_cultivation_tax",
+				args: {
+					doc: this.frm.doc
+				},
+				callback: (r) => {
+					if (r.message) {
+						let taxes = me.frm.doc.taxes;
+						let excise_tax_row = r.message;
+						console.log("excise_tax_row", excise_tax_row);
+						if (excise_tax_row.tax_amount > 0) {
```

Comment:
Is this the same as the excise tax logic for setting tax rows? Maybe we should commonify it